examples/machine_translation/model.py

```diff
@@ -0,0 +1,151 @@
+import random
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Encoder(nn.Module):
+    def __init__(
+        self, input_dim: int, emb_dim: int, enc_hid_dim: int, dec_hid_dim: int, dropout: float, embedding: nn.Module
+    ):
+        super().__init__()
+
+        self.input_dim = input_dim
+        self.emb_dim = emb_dim
+        self.enc_hid_dim = enc_hid_dim
+        self.dec_hid_dim = dec_hid_dim
+        self.dropout = dropout
+
+        self.embedding = embedding
+        self.rnn = nn.GRU(emb_dim, enc_hid_dim, bidirectional=True)
```

Comment:
Just for the record. I will add or switch transformer model in 2020 H2 as many OSS users require an example to show how to use transformer decoder. 
https://discuss.pytorch.org/t/nn-transformer-explaination/53175/14
https://discuss.pytorch.org/t/how-to-use-nn-transformerdecoder-at-inference-time/49484